lib/driver.js

```diff
@@ -123,6 +123,7 @@ class AndroidDriver extends BaseDriver {
 
       // get device udid for this session
       let {udid, emPort} = await helpers.getDeviceInfoFromCaps(this.opts);
+      this.opts.avd = this.opts.avd || /emulator/.test(udid);
```

Comment:
I agree. We already have `this.opts.udid` set. https://github.com/appium/appium-android-driver/blob/master/lib/driver.js#L126